<launch>
  <arg name="manager" />

  <arg name="focal_length_x_px" default="" />
  <arg name="focal_length_y_px" default="" />
  <arg name="principal_point_x_px" default="" />
  <arg name="principal_point_y_px" default="" />

  <arg name="publish_visualization_markers" default="false" />

  <arg name="camera_ns" doc="namespace in which camera's image_raw topic exists e.g. /usb_cam" />
  <arg name="image_topic" default="/image_raw" />
  <arg name="tag_size_cm" doc="length of the edge of the black frame in cms" />

  <param name="tag_size_cm" value="$(arg tag_size_cm)" /> 
  <node pkg="nodelet" name="april_tag" type="nodelet"
        args="load april_tag/AprilTagNodelet $(arg manager)" output="screen">
    <param name="camera_ns" value="$(arg camera_ns)" />
    <param name="image_topic" value="$(arg image_topic)" />

    <param name="focal_length_x_px" value="$(arg focal_length_x_px)" />
    <param name="focal_length_y_px" value="$(arg focal_length_y_px)" />
    <param name="principal_point_x_px" value="$(arg principal_point_x_px)" />
    <param name="principal_point_y_px" value="$(arg principal_point_y_px)" />

    <param name="publish_visualization_markers" value="$(arg publish_visualization_markers)" />
  </node>
</launch>
